### PR TITLE
Bugfix of NPE in complex binary expressions in long line rule

### PR DESCRIPTION
**Pull request by kgevorkyan**
### What's done:
* Fix NPE in long binary expressions: previously in complex cases `binList` could hold incorrect nodes, which had no `binary expression` as first common parent
* Fix out of range error in long string templates
* Add tests

### CODE REVIEW

**Comment by kgevorkyan** [comment]
Despite the fact, that for now we get rid of NPE and cover this case by tests, we still have issue on related functionality https://github.com/cqfn/diKTat/issues/984

I think, we can consider it in future PR

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt [hunk 1]
```diff
@@ -429,19 +433,33 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
...
                 } else {
-                    binList.add(it)
+                    binList.tryAdd(it)
                 }
             }
             dfsForProperty(it, binList)
         }
     }
 
+    private fun MutableList<ASTNode>.tryAdd(node: ASTNode) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt:457]:
```suggestion
    private fun MutableList<ASTNode>.addIfAbsent(node: ASTNode) {
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt [hunk 2]
```diff
@@ -429,19 +433,33 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
...
+     * @param node target node to be processed
+     * @param binList where to store the corresponding results
+     */
     private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
         node.getChildren(null).forEach {
             if (it.elementType in propertyList) {
-                if (it.elementType == REFERENCE_EXPRESSION && it.treeParent.elementType == CALL_EXPRESSION) {
-                    binList.add(it.treeParent)
+                if (it.elementType == REFERENCE_EXPRESSION && it.treeParent?.elementType == CALL_EXPRESSION) {
+                    binList.tryAdd(it.treeParent)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt:448]: Did we add the same element multiple times here? How could it be?

Comment (kgevorkyan) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt]: Yeah, thanks, I did a lot of experiments with `dfsForProperty` and it's remain from them, now it's not necessary, removed.

**Review by petertrr — APPROVED** [review]

**Pull request merged**